src/FSharp.Core/array.fs

```diff
@@ -1588,6 +1587,30 @@ module Array =
 
         acc
 
+    [<CompiledName("Sum")>]
+    let inline sum (array: ^T array) : ^T =
+        classicSum array
+        when ^T : float = 
+            if System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.StartsWith ".NET Framework" then classicSum array
```

Comment:
This should work.
Can you store it in a static boolean flag somewhere, is it possible?